### PR TITLE
Torch vector features

### PR DESCRIPTION
**Pull request by ShreyaR**
Migrates vector features to PyTorch

### CODE REVIEW

File: ludwig/data/preprocessing.py [hunk 1]
```diff
@@ -52,7 +52,7 @@
 from ludwig.utils.misc_utils import (get_from_registry, merge_dict,
                                      resolve_pointers, set_random_seed,
                                      get_proc_features_from_lists)
-from ludwig.utils.type_utils import Column
+# from ludwig.utils.type_utils import Column
```
Comment (ShreyaR) [inline ludwig/data/preprocessing.py:55]: I was running into an issue with importing Column. Will uncomment in a subsequent patch

File: ludwig/features/vector_feature.py [hunk 2]
```diff
@@ -102,26 +97,26 @@ def add_feature_data(
             raise
 
         # Determine vector size
-        vector_size = backend.df_engine.compute(
+        output_size = backend.df_engine.compute(
```
Comment (w4nderlust) [inline ludwig/features/vector_feature.py:100]: I'm not 100% sold on this choice of renaming. The way I see it `vector_size` as a name works well for bot input and output features, while `output_size` doesn't really work for the input features. Maybe I'm missing something though.

File: ludwig/modules/metric_modules.py [hunk 3]
```diff
@@ -166,7 +167,7 @@ def __init__(self, name='r2_score'):
         self.add_state("sum_y_y_hat", default=torch.tensor(0, dtype=torch.float32))
         self.add_state("N", default=torch.tensor(0, dtype=torch.float32))
 
-    def update(self, y_hat, y):
+    def update(self, y_hat: Dict[str, torch.Tensor], y: torch.Tensor):
```
Comment (w4nderlust) [inline ludwig/modules/metric_modules.py:170]: this change would likely affect all the feature using this metric, namely the numerical ones. Should they be modified to pass the dictionary too?

File: ludwig/features/base_feature.py [hunk 4]
```diff
@@ -220,7 +215,7 @@ def update_metrics(self, targets, predictions):
                 metric_fn.update(predictions, targets)
             else:
                 #metric_fn.update_state(targets, predictions[PREDICTIONS])
-                metric_fn.update(predictions[PREDICTIONS], targets)
+                metric_fn.update(predictions, targets)
```
Comment (w4nderlust) [inline ludwig/features/base_feature.py:218]: This is in the base feature, so should work across the baord for all features and all metrics

File: ludwig/features/vector_feature.py (hunk 2 above)
Comment (ShreyaR) [reply ludwig/features/vector_feature.py:100]: Hmm I see what you mean. The issue on the output vector features that requires renaming is that these parameters are passed directly to the decoder object (Projector class in this case) also used by other features. I'll think about how to fix this and upload another patch.

File: ludwig/modules/metric_modules.py (hunk 3 above)
Comment (ShreyaR) [reply ludwig/modules/metric_modules.py:170]: Currently the metric computation seems to be very inconsistent. A lot of metrics assume a dictionary of tensors as an argument while others assume a tensor. The update metrics function is also called from the `base_features` so consistency in argument types across metrics would be important

**Comment by ShreyaR** [comment]
@w4nderlust fixed the issue with `vector_size`. Per our discussion, I've kept the changes to the metric modules. Once we have consensus on next steps for unifying metric modules I can make changes to all metrics in Ludwig to be consistent.

File: tests/integration_tests/test_simple_features.py [hunk 5]
```diff
@@ -162,18 +162,18 @@ def test_feature(input_test_feature, output_test_feature,
...
+        ([category_feature()],
+         [numerical_feature(), numerical_feature()]),
+        ([category_feature()],
+         [sequence_feature(vocab_size=5), sequence_feature(vocab_size=7)]),
+        ([category_feature()],
+         [set_feature(vocab_size=5), set_feature(vocab_size=7)]),
+        ([category_feature()],
+         [text_feature(vocab_size=5), text_feature(vocab_size=7)]),
         ([category_feature()],
          [vector_feature(), vector_feature()]),
```
Comment (w4nderlust) [inline tests/integration_tests/test_simple_features.py]: this looks great for testing output vector features, although it doesn't seem to me we are testing for input vector features, or ar we?

File: ludwig/data/preprocessing.py (hunk 1 above)
Comment (ShreyaR) [reply ludwig/data/preprocessing.py]: Fixed in the latest patch

File: ludwig/features/vector_feature.py (hunk 2 above)
Comment (ShreyaR) [reply ludwig/features/vector_feature.py]: Resolved offline

File: ludwig/modules/metric_modules.py (hunk 3 above)
Comment (ShreyaR) [reply ludwig/modules/metric_modules.py:170]: Resolved offline

File: ludwig/features/base_feature.py (hunk 4 above)
Comment (ShreyaR) [reply ludwig/features/base_feature.py:218]: Resolved offline
